// antd-mobile/lib/button/index.js
'use strict';Object.defineProperty(exports,"__esModule",{value:true});exports["default"]=undefined;var _extends2=require('babel-runtime/helpers/extends');var _extends3=_interopRequireDefault(_extends2);var _classCallCheck2=require('babel-runtime/helpers/classCallCheck');var _classCallCheck3=_interopRequireDefault(_classCallCheck2);var _possibleConstructorReturn2=require('babel-runtime/helpers/possibleConstructorReturn');var _possibleConstructorReturn3=_interopRequireDefault(_possibleConstructorReturn2);var _inherits2=require('babel-runtime/helpers/inherits');var _inherits3=_interopRequireDefault(_inherits2);








var _react=require('react');var _react2=_interopRequireDefault(_react);
var _reactNative=require('react-native');
var _index=require('./style/index');var _index2=_interopRequireDefault(_index);function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{"default":obj};}var __rest=undefined&&undefined.__rest||function(s,e){var t={};for(var p in s){if(Object.prototype.hasOwnProperty.call(s,p)&&e.indexOf(p)<0)t[p]=s[p];}if(s!=null&&typeof Object.getOwnPropertySymbols==="function")for(var i=0,p=Object.getOwnPropertySymbols(s);i<p.length;i++){if(e.indexOf(p[i])<0)t[p[i]]=s[p[i]];}return t;};var
Button=function(_React$Component){(0,_inherits3["default"])(Button,_React$Component);
function Button(props){(0,_classCallCheck3["default"])(this,Button);var _this=(0,_possibleConstructorReturn3["default"])(this,
_React$Component.call(this,props));
_this.onPressIn=function(){for(var _len=arguments.length,arg=Array(_len),_key=0;_key<_len;_key++){arg[_key]=arguments[_key];}
if(!_this.props.disabled){
_this.setState({pressIn:true});
}
if(_this.props.onPressIn){
_this.props.onPressIn(arg);
}
};
_this.onPressOut=function(){for(var _len2=arguments.length,arg=Array(_len2),_key2=0;_key2<_len2;_key2++){arg[_key2]=arguments[_key2];}
if(!_this.props.disabled){
_this.setState({pressIn:false});
}
if(_this.props.onPressOut){
_this.props.onPressOut(arg);
}
};
_this.onShowUnderlay=function(){for(var _len3=arguments.length,arg=Array(_len3),_key3=0;_key3<_len3;_key3++){arg[_key3]=arguments[_key3];}
if(!_this.props.disabled){
_this.setState({touchIt:true});
}
if(_this.props.onShowUnderlay){
_this.props.onShowUnderlay(arg);
}
};
_this.onHideUnderlay=function(){for(var _len4=arguments.length,arg=Array(_len4),_key4=0;_key4<_len4;_key4++){arg[_key4]=arguments[_key4];}
if(!_this.props.disabled){
_this.setState({touchIt:false});
}
if(_this.props.onHideUnderlay){
_this.props.onHideUnderlay(arg);
}
};
_this.state={
pressIn:false,
touchIt:false};return _this;

}Button.prototype.
render=function render(){


var _a=this.props,_a$size=_a.size,size=_a$size===undefined?'large':_a$size,_a$type=_a.type,type=_a$type===undefined?'default':_a$type,disabled=_a.disabled,activeStyle=_a.activeStyle,onClick=_a.onClick,style=_a.style,styles=_a.styles,restProps=__rest(_a,["size","type","disabled","activeStyle","onClick","style","styles"]);
var textStyle=[
styles[size+'RawText'],
styles[type+'RawText'],
disabled&&styles.disabledRawText,
this.state.pressIn&&styles[type+'HighlightText']];

var wrapperStyle=[
styles.wrapperStyle,
styles[size+'Raw'],
styles[type+'Raw'],
disabled&&styles.disabledRaw,
this.state.pressIn&&activeStyle&&styles[type+'Highlight'],
activeStyle&&this.state.touchIt&&activeStyle,
style];

var underlayColor=_reactNative.StyleSheet.flatten(styles[activeStyle?type+'Highlight':type+'Raw']).backgroundColor;
return _react2["default"].createElement(_reactNative.TouchableHighlight,(0,_extends3["default"])({activeOpacity:1,delayPressOut:1,underlayColor:underlayColor,style:wrapperStyle,onPress:function onPress(e){return onClick&&onClick(e);},onPressIn:this.onPressIn,onPressOut:this.onPressOut,onShowUnderlay:this.onShowUnderlay,onHideUnderlay:this.onHideUnderlay,disabled:disabled},restProps),
_react2["default"].createElement(_reactNative.Text,{style:textStyle},this.props.children));

};return Button;}(_react2["default"].Component);exports["default"]=Button;

Button.defaultProps={
pressIn:false,
disabled:false,
activeStyle:{},
onClick:function onClick(_x){
},
onPressIn:function onPressIn(_x){
},
onPressOut:function onPressOut(_x){
},
onShowUnderlay:function onShowUnderlay(_x){
},
onHideUnderlay:function onHideUnderlay(_x){
},
styles:_index2["default"]};module.exports=exports['default'];
