// antd-mobile/lib/tabs/index.web.js
'use strict';Object.defineProperty(exports,"__esModule",{value:true});var _extends2=require('babel-runtime/helpers/extends');var _extends3=_interopRequireDefault(_extends2);var _defineProperty2=require('babel-runtime/helpers/defineProperty');var _defineProperty3=_interopRequireDefault(_defineProperty2);







var _react=require('react');var _react2=_interopRequireDefault(_react);
var _rcTabs=require('rc-tabs');var _rcTabs2=_interopRequireDefault(_rcTabs);
var _classnames=require('classnames');var _classnames2=_interopRequireDefault(_classnames);
var _SwipeableTabContent=require('rc-tabs/lib/SwipeableTabContent');var _SwipeableTabContent2=_interopRequireDefault(_SwipeableTabContent);
var _TabContent=require('rc-tabs/lib/TabContent');var _TabContent2=_interopRequireDefault(_TabContent);
var _InkTabBar=require('rc-tabs/lib/InkTabBar');var _InkTabBar2=_interopRequireDefault(_InkTabBar);
var _objectAssign=require('object-assign');var _objectAssign2=_interopRequireDefault(_objectAssign);function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{"default":obj};}var __assign=undefined&&undefined.__assign||Object.assign||function(t){for(var s,i=1,n=arguments.length;i<n;i++){s=arguments[i];for(var p in s){if(Object.prototype.hasOwnProperty.call(s,p))t[p]=s[p];}}return t;};
var Tabs=_react2["default"].createClass({displayName:'Tabs',
statics:{
TabPane:_rcTabs.TabPane},

getDefaultProps:function getDefaultProps(){
return{
prefixCls:'am-tabs',
animated:true,
swipeable:true,
tabBarPosition:'top',
hammerOptions:{},
onChange:function onChange(){},
onTabClick:function onTabClick(){}};

},
getInitialState:function getInitialState(){var _props=
this.props,activeKey=_props.activeKey,defaultActiveKey=_props.defaultActiveKey,children=_props.children;
var activeTabKey=activeKey||defaultActiveKey||children[0].props.key;
var activeTabIndex=children.findIndex(function(tabPane){return tabPane.key===activeTabKey;});
return{

viewportStartTabIndex:children.length>5&&activeTabIndex>=3?activeTabIndex-2:0};

},
renderTabBar:function renderTabBar(){var
props=this.props;
return _react2["default"].createElement(_InkTabBar2["default"],{onTabClick:this.handleTabClick,inkBarAnimated:props.animated});
},
renderTabContent:function renderTabContent(){var _props2=
this.props,animated=_props2.animated,swipeable=_props2.swipeable,hammerOptions=_props2.hammerOptions;
return swipeable?_react2["default"].createElement(_SwipeableTabContent2["default"],{animated:animated,hammerOptions:hammerOptions}):_react2["default"].createElement(_TabContent2["default"],{animated:animated});
},
getChildren:function getChildren(){var
children=this.props.children;var
viewportStartTabIndex=this.state.viewportStartTabIndex;
return children.slice(viewportStartTabIndex,viewportStartTabIndex+5);
},
handleTabClick:function handleTabClick(key){
this.setState({
viewportStartTabIndex:this.getStartTabIndex(key)});

this.props.onTabClick(key);
},
handleTabChange:function handleTabChange(key){
this.setState({
viewportStartTabIndex:this.getStartTabIndex(key)});

this.props.onChange(key);
},
getStartTabIndex:function getStartTabIndex(activeTabKey){var
children=this.props.children;var
viewportStartTabIndex=this.state.viewportStartTabIndex;
var activeTabIndex=children.findIndex(function(tabPane){return tabPane.key===activeTabKey;});

if(children.length<5){
return 0;
}

if(activeTabIndex===viewportStartTabIndex&&viewportStartTabIndex>0){
return viewportStartTabIndex-1;
}

if(activeTabIndex===viewportStartTabIndex+4&&viewportStartTabIndex+4<children.length-1){
return viewportStartTabIndex+1;
}

return viewportStartTabIndex;
},
getClassName:function getClassName(){var _className;var
viewportStartTabIndex=this.state.viewportStartTabIndex;
var totalTabsCount=this.props.children.length;
var cls=(0,_classnames2["default"])((_className={},(0,_defineProperty3["default"])(_className,
this.props.prefixCls+'-leftpage',totalTabsCount>5&&viewportStartTabIndex>0),(0,_defineProperty3["default"])(_className,
this.props.prefixCls+'-rightpage',totalTabsCount>5&&viewportStartTabIndex+4<totalTabsCount-1),_className));

return cls;
},
render:function render(){
var newProps=__assign({},this.props,{onChange:this.handleTabChange,children:this.getChildren(),className:(0,_objectAssign2["default"])(this.getClassName(),this.props.className)});
newProps.className = "lll";
return _react2["default"].createElement(_rcTabs2["default"],(0,_extends3["default"])({renderTabBar:this.renderTabBar,renderTabContent:this.renderTabContent},newProps));
}});exports["default"]=

Tabs;module.exports=exports['default'];
